lib/sidebar_builder.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: false
+
+module SwedbankPay
+  # A nice sidebar
+  class SidebarBuilder
+    def initialize(filename, pages)
+      @filename = filename
+      @pages = pages
+    end
+
+    def build
+      sidebar_markup = ''
+
+      @pages.select { |path, _| path.split('/').length <= 2 }.each do |path, page|
+        next if page[:title].nil?
+        next if page[:hide_from_sidebar]
+
+        sidebar_markup << generate_main_markup(path, page)
+      end
+
+      sidebar_markup
+    end
+
+    private
+
+    def generate_main_markup(path, page)
+      title = page[:title].split('–').first
+      child_pages = @pages.select { |child_path, _| child_path.include?(path) && child_path != path && path != '/' }
+      item_class = item_class(path)
+      child_markup = generate_child_markup(path, page, child_pages, 2)
+      grandchildren_markup = generate_grandchildren_markup(child_pages)
+
+      "<li class=\"#{item_class}\">
+        <div class=\"nav-group-heading\">
+          <i class=\"material-icons\">arrow_right</i>
+          <span>#{title}</span>
+        </div>
+        <ul class=\"nav-ul\">
+          #{child_markup}
+          #{grandchildren_markup}
+        </ul>
+      </li>"
+    end
+
+    def item_class(path)
+      active = @filename.active?(path)
+      return 'nav-group active' if active
+
+      'nav-group'
+    end
+
+    def generate_child_markup(path, page, all_child_pages, level)
```

Comment:
Codacy found an issue: [Cyclomatic complexity for generate_child_markup is too high. [13/6]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)